src/block_cache/block_io.c

```diff
@@ -477,16 +591,26 @@ __wt_blkcache_write(WT_SESSION_IMPL *session, WT_ITEM *buf, WT_PAGE_BLOCK_META *
 
         WT_ERR(bm->write_size(bm, session, &size));
         WT_ERR(__wt_scr_alloc(session, size, &etmp));
-        WT_ERR(__wt_encrypt(session, kencryptor, WT_BLOCK_ENCRYPT_SKIP, ip, etmp));
+        WT_ASSERT(session, ip->size > 0);
+        WT_ERR(__wt_encrypt(session, kencryptor,
+          bm->encrypt_skip(bm, session, block_meta->delta_count > 0), ip, etmp));
 
         encrypted = true;
         ip = etmp;
 
         /* Set the disk header flags. */
-        dsk = ip->mem;
-        if (compressed)
-            F_SET(dsk, WT_PAGE_COMPRESSED);
-        F_SET(dsk, WT_PAGE_ENCRYPTED);
+        if (block_meta->delta_count != 0) {
```

Comment:
May be we should use a local variable for this instead of checking it everytime.

Reply:
Good call, done in ab8121d7b9ccd38510fb1b627f705cf30ab03f72.